lib/chef/resource/mdadm.rb

```diff
@@ -99,6 +106,37 @@ def raid_device(arg=nil)
         )
       end
 
+      # Can be removed once metadata and chunk defaults are set to nil
+      def user_set_metadata(arg=nil)
+        set_or_return(
+          :exists,
+          arg,
+          :kind_of => [ TrueClass, FalseClass ]
+        )
+      end
+
+      # Can be removed once metadata and chunk defaults are set to nil
+      def user_set_chunk(arg=nil)
+        set_or_return(
+          :exists,
+          arg,
+          :kind_of => [ TrueClass, FalseClass ]
+        )
+      end
+
+      # Can be removed once metadata and chunk defaults are set to nil
+      def after_created
+        metadata_warn_msg = 'The default metadata version of 0.90 will be '\
+          'removed in a future release. This change will allow mdadm to '\
+          'choose the default metadata version.'
+        chunk_warn_msg = 'The default chunk size of 16k will be removed in a '\
+          'future release.  This change will allow mdadm to choose '\
+          'the default chunk size.'
```

Comment:
Cool, yep, wordsmith it how you like.  I was mostly concerned that it not only emit the warning but make sure that the reader knows what action to take next.